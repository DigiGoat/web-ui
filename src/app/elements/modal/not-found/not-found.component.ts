import { Component, Input } from '@angular/core';

@Component({
    selector: 'app-modal-not-found',
    templateUrl: './not-found.component.html',
    styleUrl: './not-found.component.scss',
    standalone: false
})
export class NotFoundComponent {
  @Input({ required: true }) searchParam!: string;
}
