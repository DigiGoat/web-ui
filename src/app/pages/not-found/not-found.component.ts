import { Component, type OnDestroy, type OnInit } from '@angular/core';
import { Meta } from '@angular/platform-browser';
import type { Observable } from 'rxjs';
import type { Page } from '../../app-routing.module';

@Component({
    selector: 'app-not-found',
    templateUrl: './not-found.component.html',
    styleUrl: './not-found.component.scss',
    standalone: false
})
export class NotFoundComponent implements OnInit, Page, OnDestroy {
  constructor(private meta: Meta) { }
  ngOnInit(): void {
    this.setDescription();
  }
  setDescription(): void | Observable<void> {
    this.meta.addTag({ name: 'robots', content: 'noindex' });
  }
  ngOnDestroy(): void {
    this.meta.removeTag('name="robots"');
  }
}
