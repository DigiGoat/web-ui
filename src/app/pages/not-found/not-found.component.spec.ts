import { ComponentFixture, TestBed } from '@angular/core/testing';

import type { Meta } from '@angular/platform-browser';
import { NotFoundComponent } from './not-found.component';

describe('NotFoundComponent', () => {
  let component: NotFoundComponent;
  let fixture: ComponentFixture<NotFoundComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [NotFoundComponent]
    })
      .compileComponents();

    fixture = TestBed.createComponent(NotFoundComponent);
    component = fixture.componentInstance;
    component['meta'] = { addTag: jest.fn(), removeTag: jest.fn() } as unknown as Meta;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should add a `noindex` tag', () => {
    expect(component['meta'].addTag).toHaveBeenCalledTimes(1);
    expect(component['meta'].addTag).toHaveBeenCalledWith({ name: 'robots', content: 'noindex' });
  });
  it('should remove a `noindex` tag when destroyed', () => {
    fixture.destroy();
    expect(component['meta'].removeTag).toHaveBeenCalledTimes(1);
    expect(component['meta'].removeTag).toHaveBeenCalledWith('name="robots"');
  });
  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
