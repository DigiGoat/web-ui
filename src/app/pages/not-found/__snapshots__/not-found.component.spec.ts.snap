// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundComponent should match the snapshot 1`] = `
<app-not-found
  meta={[Function Object]}
>
  <h1
    class="text-center"
  >
    Page Not Found
  </h1><h5
    class="text-center"
  >
    Please check your URL and try again
  </h5>
</app-not-found>
`;
